apps/webpack.base.config.js

```diff
@@ -0,0 +1,249 @@
+          {
+            loader: 'css-loader',
+            options: {
+              modules: {
+                auto: true,
+                localIdentName: process.env.DEV
+                  ? '[path][name]__[local]'
+                  : '[hash:base64]',
+              },
+            },
+          },
+          {
+            loader: 'sass-loader',
+            options: {
+              implementation: sass,
+              sassOptions: {
+                includePaths: [p('../shared/css')],
+                outputStyle: 'compressed',
+              },
+            },
+          },
+        ],
+      },
+
+      {test: /\.interpreted.js$/, type: 'asset/source'},
+      {
+        test: /\.(png|jpg|jpeg|gif|svg)$/,
+        include: [
+          p('static'),
+          p('src'),
+          p('test'),
+          p('../dashboard/app/assets/images'),
+        ],
+        // note that in the name template given below, a dash prefixing
+        // the hash is explicitly avoided. If rails tries to serve
+        // this file when asset digests are turned off, it will return a
+        // 404 because it thinks the hash is a digest and it won't
+        // be able to find the file without the hash. :( :(
+        use: [
+          {
+            loader: 'url-loader',
+            options: {
+              limit: 1024,
+              // uses the file-loader when file size is over the limit
+              name: '[name]wp[contenthash].[ext]',
+              esModule: false,
+            },
+          },
+        ],
+      },
+      {
+        test: /\.jsx?$/,
+        enforce: 'pre',
+        include: [...nodeModulesToTranspile, p('src'), p('test')],
+        exclude: [p('src/lodash.js')],
+        loader: 'babel-loader',
+        options: {
+          cacheDirectory: p('.babel-cache'),
+          compact: false,
+        },
+      },
+      {
+        test: /\.tsx?$/,
+        use: 'ts-loader',
+        exclude: /node_modules/,
+      },
+      // modify baseConfig's preLoaders if looking for code coverage info
+      ...(envConstants.COVERAGE
+        ? [
+            {
+              test: /\.jsx?$/,
+              enforce: 'post',
+              loader: 'istanbul-instrumenter-loader',
+              include: p('src'),
+              exclude: [
+                // we need to turn off instrumentation for this file
+                // because we have tests that actually make assertions
+                // about the contents of the compiled version of this file :(
+                p('src/flappy/levels.js'),
+              ],
+              options: {
+                cacheDirectory: true,
+                compact: false,
+                esModules: true,
+              },
+            },
+          ]
+        : []),
+    ],
+    noParse: [/html2canvas/],
+  },
+};
+
+// if (envConstants.HOT) {
+//   baseConfig.module.loaders.push({
+//     test: /\.jsx?$/,
+//     loader: 'react-hot-loader',
+//     include: [p('src')],
+//   });
+// }
```

Comment:
I believe this code cannot work as it was written, so I've commented it out for now. If envConstants.HOT is set to true, webpack refuses to continue since `config.module.loaders` hasn't been a valid webpack config property for a long time. If folks confirm that nobody is using envConstants.HOT, I'll happily delete this block (and maybe drop it from envConstants.HOT too? 🤷‍♂️).

Does react hot loading work by some other means? If not, is that something we'd like to have?

Reply:
IIRC this just stopped working during a past upgrade, maybe webpack 2 to 3 or so. hot reloading would be nice to have. based on the fact that some folks choose to develop react components in storybook for faster iteration, I'd say hot reloading would be popular with anyone working on react components

Reply:
Aha! That tracks, since it /used/ to be a valid webpack property, but was deprecated and then removed. Since I'm doing more work in this area, I'll take a short at getting hot reload working and do a followup PR if I can get it going again.